salt/modules/pkgng.py

```diff
@@ -74,10 +75,10 @@ def __virtual__():
         providers = {}
         if 'providers' in __opts__:
             providers = __opts__['providers']
-        log.debug('__opts__.providers: {0}'.format(providers))
+        log.debug('__opts__.providers: %s', str(providers))
```

Comment:
There is no need to use `str()` here. The fact that you have used a `%s` placeholder in the log string means that Python's logging module will convert the dictionary to a string.

Reply:
Thank you for the code review, @terminalmage. The last item is fixed.